ios/StripeTerminalReactNative.swift

```diff
@@ -420,7 +427,36 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
             return
         }
 
-        Terminal.shared.createPaymentIntent(paymentParams) { pi, error in
+        let offlineBehavior = params["offlineBehavior"] as? String
+        let offlineModeTransactionLimit = params["offlineModeTransactionLimit"] as? NSNumber ?? 0
+        let offlineModeStoredTransactionLimit = params["offlineModeStoredTransactionLimit"] as? NSNumber ?? 0
+        
+        var isOverOfflineTransactionLimit = amount.intValue >= offlineModeTransactionLimit.intValue
+        if let offlinePaymentTotalByCurrency = Terminal.shared.offlineStatus.sdk.paymentAmountsByCurrency[paymentParams.currency]?.intValue {
+            isOverOfflineTransactionLimit = isOverOfflineTransactionLimit || (offlinePaymentTotalByCurrency >= offlineModeStoredTransactionLimit.intValue)
+        }
+        let offlineBehaviorFromTransactionLimit: OfflineBehavior = {
+            if isOverOfflineTransactionLimit {
+                return .requireOnline
+            } else {
```

Comment:
same as above, we can remove this logic and just directly pass whatever OfflineBehavior user passed to the SDK